src/migrations/20250423092622-add-standard-goal-constraint.js

```diff
@@ -0,0 +1,24 @@
+const { prepMigration } = require('../lib/migration');
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface) {
+    await queryInterface.sequelize.transaction(async (transaction) => {
+      const sessionSig = __filename;
+      await prepMigration(queryInterface, transaction, sessionSig);
+      // Add a index constraint to the Goals table to ensure there is only one combination
+      // of "grantId" and "goalTemplateId" where the "status" is not equal to 'Closed'.
+      await queryInterface.sequelize.query(`
+                CREATE UNIQUE INDEX IF NOT EXISTS "unique_grantId_goalTemplateId" ON "Goals" ("grantId", "goalTemplateId")
+                WHERE "status" != 'Closed' AND "goalTemplateId" IS NOT NULL AND "deletedAt" IS NULL;
```

Comment:
I think excluding deleted records is fine, especially because we're not *removing* any index to make room for this, so even if we do have something somewhere looking at deleted records, it's performing okay with the indexing we have now.

Reply:
@hardwarehuman to confirm, we are keeping it as is for now right?